src/number-util/number-util.ts

```diff
@@ -0,0 +1,23 @@
+export namespace NumberUtil {
+  export function intValueOf(numStr: string): number {
+    const result = isNumeric(numStr) ? Number(numStr) : NaN;
+
+    if (Number.isNaN(result) || !Number.isInteger(result)) {
+      throw new Error(`unable to convert "${numStr}" to integer type`);
+    }
+    return result;
+  }
+
+  export function valueOf(numStr: string): number {
+    const result = isNumeric(numStr) ? Number(numStr) : NaN;
+
+    if (Number.isNaN(result) || !Number.isFinite(result)) {
+      throw new Error(`unable to convert "${numStr}" to integer type`);
+    }
+    return result;
```

Comment:
Number.isNaN() 는 불필요하게 두번 하고 있습니다. 2
```suggestion
    const result = Number(numStr);

    if (!numStr || !Number.isFinite(result)) {
      throw new Error(`unable to convert "${numStr}" to integer type`);
    }

    return result;
```